go/ql/lib/change-notes/2024-03-20-dependecy-retrieval-improvement.md

```diff
@@ -0,0 +1,4 @@
+---
+category: minorAnalysis
+---
+* The `CODEQL_EXTRACTOR_GO_FAST_PACKAGE_INFO` option, which speeds up retrieval of dependency information, is now on by default. This was originally an external contribution by @xhd2015.
```

Comment:
Since this option is undocumented, it may be worth mentioning that it can be turned off by setting it to false. Something like

>To disable this behaviour, set `CODEQL_EXTRACTOR_GO_FAST_PACKAGE_INFO=false`.

Reply:
Not opposed to documenting it, but two points:

- Would it be better for us to document this somewhere that's easier to find than the change notes for a particular version?
- If we document it, we should come up with a better description of what it does and why someone might want to disable it. Since I couldn't reproduce any of the failures that happened in the past, it seems like we are not currently aware of any reasons why someone wouldn't want to this to be on. We have left the option just in case we find something going forward where disabling it is beneficial.

Reply:
I am happy to leave it pretty much undocumented, and silently remove it in future if we don't hear of anyone having a need for it. So maybe your original change note is best.

Reply:
I will leave the change note as is for now then.